CDP4Common/Extensions/ParameterTypeExtensions.cs

```diff
@@ -0,0 +1,206 @@
+//    This file is part of CDP4-COMET SDK Community Edition
+// 
+//    The CDP4-COMET SDK Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Lesser General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or (at your option) any later version.
+// 
+//    The CDP4-COMET SDK Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Lesser General Public License for more details.
+// 
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with this program; if not, write to the Free Software Foundation,
+//    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+// </copyright>
+// -------------------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Common.Extensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using CDP4Common.DTO;
+
+    /// <summary>
+    /// Extension class for <see cref="ParameterType" />
+    /// </summary>
+    public static class ParameterTypeExtensions
+    {
+        /// <summary>
+        /// Retrieves inside a collection of <see cref="Thing" /> the association of a <see cref="ParameterType" />
+        /// and <see cref="MeasurementScale" /> for a given <see cref="ParameterTypeComponent" />
+        /// </summary>
+        /// <param name="component">
+        /// The <see cref="ParameterTypeComponent" /> that provides <see cref="Guid" />s to retrieve the
+        /// <see cref="ParameterType" /> and the <see cref="MeasurementScale" />, if applicable
+        /// </param>
+        /// <param name="things">
+        /// A collection of <see cref="Thing" />s to be able to retrieve referenced
+        /// <see cref="ParameterType" /> and the <see cref="MeasurementScale" />, if applicable
+        /// </param>
+        /// <returns>The <see cref="ParameterType" /> and the <see cref="MeasurementScale" /> association</returns>
+        /// <exception cref="InvalidDataException">
+        /// If the requested <see cref="ParameterType" /> or the
+        /// <see cref="MeasurementScale" /> is not present in the <paramref name="things" />
+        /// </exception>
+        public static (ParameterType parameterType, MeasurementScale scale) QueryParameterTypeAndMeasurementScale(this ParameterTypeComponent component, IReadOnlyCollection<Thing> things)
+        {
+            var parameterType = things.OfType<ParameterType>().FirstOrDefault(x => x.Iid == component.ParameterType)
+                                ?? throw new InvalidDataException($"The provided collection of Things do not contains a reference to the ParameterType {component.ParameterType}");
```

Comment:
please consider using the `ThingNotFoundException` for this case

Reply:
I think the message should be
The provided collection of Things **does not contain** a reference to the ParameterType.....